cmd/wavelet/completion.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019 Perlin
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package main
+
+import "github.com/chzyer/readline"
+
+/*
+func (cli *CLI) getCompletionPeers() *readline.PrefixCompleter {
+	fn := func(s string) (l []string) {
+		// Get a list of peers
+		peers := cli.client.ClosestPeerIDs()
+		l = make([]string, 0, len(peers))
+
+		for _, id := range peers {
+			pub := id.PublicKey()
+			l = append(l, hex.EncodeToString(pub[:]))
+		}
+
+		// Get current ID
+		publicKey := cli.keys.PublicKey()
+
+		// Get root ID
+		round := cli.ledger.Rounds().Latest()
+
+		l = append(l,
+			hex.EncodeToString(round.End.ID[:]),
+			hex.EncodeToString(publicKey[:]),
+		)
+
+		fields := strings.Fields(s)
+
+		if len(fields) > 1 {
+			l = containStrings(l, fields[1], false)
+		}
+
+		return
+	}
+
+	return readline.PcItemDynamic(fn)
+}
+*/
+
+func (cli *CLI) getCompleter() *readline.PrefixCompleter {
```

Comment:
Think this function could be inlined,

Reply:
Will cleanup after merging in the autocomplete PR though.

Reply:
If this is to be inlined, it'll look like this 
```go
	for _, cmd := range c.app.Commands {
		completers = append(completers, readline.PcItem(
			cmd.Name, readline.PcItemDynamic(func(string) []string {
				return c.completion
			}),
		))

		for _, alias := range cmd.Aliases {
			completers = append(completers, readline.PcItem(
				alias, readline.PcItemDynamic(func(string) []string {
					return c.completion
				}),
			))
		}
	}
``` 

in `cli.go`

Reply:
Gotcha; no problem though since the completers will be manually computed from `ledger.Find()`.

Reply:
Alright, I pushed a commit to inline it anyway, since it could be cleaner than the code above.